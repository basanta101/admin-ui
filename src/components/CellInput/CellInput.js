import { useEffect, useRef } from 'react'
import "./CellInput.css";

const CellInput = ({
  onChange = (f) => f,
  value = "",
  pattern = "",
  type = "text",
  onBlur = (f) => f,
  autofocus = false,
}) => {

  const inputRef = useRef();

  useEffect(() => {
   if(autofocus) {
     inputRef.current.focus();
   }
  }, []);

  return (
    <input
      aria-label="cell-input"
      ref={node => inputRef.current = node }
      className="cellInput"
      onChange={(e) => onChange(e.target.value)}
      value={value}
      type={type}
      pattern={pattern}
      onBlur={onBlur}
    />
  );
};

export default CellInput;
