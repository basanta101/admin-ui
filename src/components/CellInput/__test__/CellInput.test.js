import React from "react";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import CellInput from "../CellInput";

test("change event is triggered when user inputs", () => {
  let value = "";
  const onChange = jest.fn((val) => {
    value = val;
  });
  const { container, rerender } = render(
    <CellInput onChange={onChange} value={value}/>
  );
  const input = container.firstChild;
  fireEvent.change(input, { target: { value: "$23" } });
  expect(onChange).toHaveBeenCalledTimes(1);
  rerender(<CellInput value={value} handleChange={onChange} />);
  expect(input.value).toBe("$23");
});

test("onBlur is called when user leaves the input field", () => {
  const onBlur = jest.fn();
  const { container } = render(
    <CellInput onBlur={onBlur} />
  );
  const input = container.firstChild;
  fireEvent.blur(input);
  expect(onBlur).toHaveBeenCalledTimes(1);
});

it("renders a snapshot of CellInput", () => {
  const tree = renderer.create(<CellInput />).toJSON();
  expect(tree).toMatchSnapshot();
});