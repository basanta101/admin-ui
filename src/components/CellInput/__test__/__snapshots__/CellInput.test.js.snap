// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of CellInput 1`] = `
<input
  aria-label="cell-input"
  className="cellInput"
  onBlur={[Function]}
  onChange={[Function]}
  pattern=""
  type="text"
  value=""
/>
`;
